c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -335,6 +347,14 @@ impl<'tcx> TypeChecker<'tcx, '_> {
         match stmt.kind {
             StatementKind::Assign(ref x) => {
                 let (pl, ref rv) = **x;
+
+                if self.c_void_casts.is_special(&pl, &rv) {
+                    // skip this cast, because the local that is getting casted
+                    // originates from a call to an allocation that is handled
+                    // in a way that effectively elides the cast
```

Comment:
Can you add part of these docs to `is_special` (or whatever we call it), too?

Reply:
912688a9